### ISSUE TITLE
Bug: TypeScript: Broken plugin types

### ISSUE THREAD

**Issue by danielrentz** [vitest-dev/eslint-plugin-vitest#633]
Starting in version 1.1.21, the TypeScript types of this plugin are broken. This is noticable only if the plugin is used in TS files, and with the `typescript-eslint` plugin.

Up to 1.1.20, the following test code passes

```ts
import type { Linter } from "eslint";
import vitest from "@vitest/eslint-plugin";
const TEST: Linter.Config = {
    languageOptions: { globals: vitest.environments.env.globals },
    rules: vitest.configs.recommended.rules,
};
```

Starting with 1.1.21, the code fails due to overly incomplete/generic typings in `dist/index.d.ts`.

```
src/eslint/env/vitest.ts:18:33 - error TS18048: 'vitestPlugin.environments' is possibly 'undefined'.

18     languageOptions: { globals: vitest.environments.env.globals },
                                   ~~~~~~~~~~~~~~~~~~~
```
because suddenly `environments` became optional although it always exists.

Fixing it with a non-null assertion leads to some ESLint assertions regarding using `any` values:

- `environments` is a `Record<string, any>` thus trying to access `env.globals` leads to "error  Unsafe member access .globals on an `any` value  @typescript-eslint/no-unsafe-member-access"
- `configs` is a `Record<string, any>` thus trying to access `recommended.rules` leads to "error  Unsafe member access .rules on an `any` value  @typescript-eslint/no-unsafe-member-access"

**Comment by danielrentz** [comment]
Oh well, I just saw your 1.1.24 release two hours ago :)))
